main.py

```
@@ -80,6 +80,15 @@ async def get_activities(start: int = 0, limit: int = 100, api: Garmin = Depends
 async def get_last_activity(api: Garmin = Depends(get_garmin_api)):
     return {"last_activity": api.get_last_activity()}
 
+
+@app.get("/get_previous_day_steps")
+async def get_previous_day_steps(date: str = None, api: Garmin = Depends(get_garmin_api)):
```

Ruff (0.8.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)